packages/login/src/views/ResetCredentialsForm/AuthDetailsVerificationForm.tsx

```diff
@@ -58,245 +58,221 @@ type Props = BaseProps &
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      phone: '',
-      email: '',
-      touched: false,
-      error: false,
-      errorMessage: '',
-      notificationMethod: window.config.USER_NOTIFICATION_DELIVERY_METHOD
-    }
-  }
+const AuthDetailsVerificationComponent = ({
+  intl,
+  goToForgottenItemForm,
+  goToRecoveryCodeEntryForm,
+  goToSecurityQuestionForm,
+  location
+}: Props) => {
+  const [phone, setPhone] = useState('')
+  const [email, setEmail] = useState('')
+  const [touched, setTouched] = useState(false)
+  const [error, setError] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [notificationMethod] = useState(
+    window.config.USER_NOTIFICATION_DELIVERY_METHOD
+  )
 
-  handleMobileChange = (value: string) => {
-    this.setState({
-      error: phoneNumberFormat(value) ? true : false,
-      phone: value,
-      touched: true
-    })
+  const handleMobileChange = (value: string) => {
+    setPhone(value)
+    setTouched(true)
+    setError(phoneNumberFormat(value) ? true : false)
   }
 
-  handleEmailChange = (value: string) => {
-    this.setState({
-      error: emailAddressFormat(value) ? true : false,
-      email: value,
-      touched: true
-    })
+  const handleEmailChange = (value: string) => {
+    setEmail(value)
+    setTouched(true)
+    setError(emailAddressFormat(value) ? true : false)
   }
 
-  handleContinue = async (event: React.FormEvent) => {
+  const handleContinue = async (event: React.FormEvent) => {
     event.preventDefault()
-    if (
-      this.state.notificationMethod === 'sms' &&
-      (!this.state.phone || this.state.error)
-    ) {
-      this.setState((prevState) => ({
-        touched: true,
-        error: true,
-        errorMessage: !prevState.phone
-          ? this.props.intl.formatMessage(validationMessages.phoneNumberFormat)
-          : this.props.intl.formatMessage(messages.errorPhoneNumberNotFound)
-      }))
+
+    if (notificationMethod === 'sms' && (!phone || error)) {
+      setError(true)
+      setErrorMessage(
+        !phone
+          ? intl.formatMessage(validationMessages.phoneNumberFormat)
+          : intl.formatMessage(messages.errorPhoneNumberNotFound)
+      )
       return
-    } else if (
-      this.state.notificationMethod === 'email' &&
-      (!this.state.email || this.state.error)
-    ) {
-      this.setState((prevState) => ({
-        touched: true,
-        error: true,
-        errorMessage: !prevState.email
-          ? this.props.intl.formatMessage(validationMessages.emailAddressFormat)
-          : this.props.intl.formatMessage(messages.errorEmailAddressNotFound)
-      }))
+    }
+    if (notificationMethod === 'email' && (!email || error)) {
+      setError(true)
+      setErrorMessage(
+        !email
+          ? intl.formatMessage(validationMessages.emailAddressFormat)
+          : intl.formatMessage(messages.errorEmailAddressNotFound)
+      )
```

Comment:
We should make `touched: true` like how it used to be done previously

Reply:
Glad you noticed that! And thanks for reviewing!🤗